pgmonitor_extension/sql/matviews/matviews.sql

```diff
@@ -0,0 +1,211 @@
+INSERT INTO @extschema@.metric_views (
+    view_name 
+    , run_interval
+    , scope )
+VALUES (
+   'ccp_locks'
+    , '1 minute'::interval
+    , 'global');
+
+
+CREATE MATERIALIZED VIEW @extschema@.ccp_stat_bgwriter AS
+    SELECT checkpoints_timed
+    , checkpoints_req
+    , checkpoint_write_time
+    , checkpoint_sync_time
+    , buffers_checkpoint
+    , buffers_clean
+    , maxwritten_clean
+    , buffers_backend
+    , buffers_backend_fsync
+    , buffers_alloc
+    , stats_reset 
+    FROM pg_catalog.pg_stat_bgwriter;
+/* According to docs, this table should only ever have 1 row */
+CREATE UNIQUE INDEX ccp_stat_bgwriter_idx ON @extschema@.ccp_stat_bgwriter (stats_reset);
+INSERT INTO @extschema@.metric_views (
+    view_name 
+    , run_interval
+    , scope 
+    , concurrent_refresh )
+VALUES (
+   'ccp_stat_bgwriter'
+    , '5 minutes'::interval
+    , 'global'
+    , false );
+
+
+CREATE MATERIALIZED VIEW @extschema@.ccp_stat_database AS
+    SELECT s.datname AS dbname
+    , xact_commit
+    , xact_rollback
+    , blks_read
+    , blks_hit
+    , tup_returned
+    , tup_fetched
+    , tup_inserted
+    , tup_updated
+    , tup_deleted
+    , conflicts
+    , temp_files
+    , temp_bytes
+    , deadlocks
+    FROM pg_catalog.pg_stat_database s 
+    JOIN pg_catalog.pg_database d ON d.datname = s.datname
+    WHERE d.datistemplate = false;
+CREATE UNIQUE INDEX ccp_stat_database_idx ON @extschema@.ccp_stat_database (dbname);
+INSERT INTO @extschema@.metric_views (
+    view_name 
+    , run_interval
+    , scope )
+VALUES (
+   'ccp_stat_database'
+    , '5 minutes'::interval
+    , 'global');
+
+
+CREATE MATERIALIZED VIEW @extschema@.ccp_sequence_exhaustion AS
+    SELECT count FROM @extschema@.sequence_exhaustion(75);
+CREATE UNIQUE INDEX ccp_sequence_exhaustion_idx ON @extschema@.ccp_sequence_exhaustion (count);
+INSERT INTO @extschema@.metric_views (
+    view_name 
+    , run_interval
+    , scope )
+VALUES (
+   'ccp_sequence_exhaustion'
+    , '5 minutes'::interval
+    , 'database');
+
+
+CREATE MATERIALIZED VIEW @extschema@.ccp_pg_settings_checksum AS
+    SELECT @extschema@.pg_settings_checksum() AS status;
+CREATE UNIQUE INDEX ccp_pg_settings_checksum_idx ON @extschema@.ccp_pg_settings_checksum (status);
+INSERT INTO @extschema@.metric_views (
+    view_name 
+    , run_interval
+    , scope )
+VALUES (
+   'ccp_pg_settings_checksum'
+    , '5 minutes'::interval
+    , 'global');
+
+
+CREATE MATERIALIZED VIEW @extschema@.ccp_pg_hba_checksum AS
+    SELECT @extschema@.pg_hba_checksum() AS status;
+CREATE UNIQUE INDEX ccp_pg_hba_checksum_idx ON @extschema@.ccp_pg_hba_checksum (status);
+INSERT INTO @extschema@.metric_views (
+    view_name 
+    , run_interval
+    , scope )
+VALUES (
+   'ccp_pg_hba_checksum'
+    , '5 minutes'::interval
+    , 'global');
+
+-- TODO add ccp_data_checksum_failure that calls function that returns NULL even on PG11.
+
+CREATE MATERIALIZED VIEW @extschema@.ccp_wal_activity AS
```

Comment:
This may be different on replicas. Check this and other metrics to see which may need to be switched to being views to allow primary vs replica differences

Reply:
is this still a concern? or should this convo be resolved @keithf4 ?

Reply:
Will review this again. thanks

Reply:
Also, this extension has been moved to its own repo, so this file is no longer valid here